### DISCUSSION TITLE
Is it possible to do data driven testing with external data source

### DISCUSSION THREAD

**Discussion by javawithrocky** [cucumber/common#2209] [Q&A Java]
Hello,

I am curious if Cucumber JVM allows data driven tests such that data is driven from outside feature file. For example: let's say I want to test a function that sums 2 integers. I want to provide a List of 3 integers (first 2 are input, and 3rd is expected result). These numbers are stored in a database. So I want my feature file to look something like this:

```
Scenario: Verify sum
  Given that we want to sum <a> and <b>
  When we perform sum
  Then expected result is <c>
```

I don't want the data to live in the feature file but be programmatically be plugged in from say a database. Let's say this database table has 3 rows. Then I want to execute above `Scenario` 3 times.

Is it possible?

Thanks in adance.

**Comment by mpkorstanje** [comment]
Cucumber JVM doesn't support that. Feature files are assumed to be self contained.
